lib/dynosaur/utils/os.rb

```diff
@@ -0,0 +1,24 @@
+# Utility methods for interacting with the underlying OS
+module Dynosaur
+  module Utils
+    module OS
+      class << self
+        # @return [Array] Object instances, which respond to #command, for all
+        #   locally running processes returned by the ps command
+        def structured_ps
+          ps_array = command_ps.split("\n").map(&:strip)
+          ps_array.shift # Remove the header row
+          ps_array.map do |command|
+            Struct.new(:command).new(command)
+          end
+        end
+
+        private
+
+        def command_ps
+          `ps -o command`
```

Comment:
Are there any OS-dependent issues we need to worry about here? We could use something like [sys-proctable](https://github.com/djberg96/sys-proctable)